### PR TITLE
Fix showothers for specchar

### PR DESCRIPTION
**Pull request by edg-l**
Fixes https://github.com/ddnet/ddnet/issues/2410

I also made it use the Alpha on specchars that are on another team.

### CODE REVIEW

File: src/game/client/components/nameplates.cpp [hunk 1]
```diff
@@ -177,7 +177,11 @@ void CNamePlates::OnRender()
 
 		if(m_pClient->m_aClients[i].m_SpecCharPresent)
 		{
-			RenderNameplatePos(m_pClient->m_aClients[i].m_SpecChar, pInfo, 0.4f);
+			bool OtherTeam = m_pClient->IsOtherTeam(i);
+			float Alpha = OtherTeam ? g_Config.m_ClShowOthersAlpha / 100.0f : 0.4f;
```
Comment (heinrich5991) [inline src/game/client/components/nameplates.cpp:181]:
```suggestion
			float Alpha = 0.4f * (OtherTeam ? g_Config.m_ClShowOthersAlpha / 100.0f : 1.0f);
```

Would be more in line with the tee rendering.

File: src/game/server/player.cpp [hunk 2]
```diff
@@ -404,7 +404,12 @@ void CPlayer::Snap(int SnappingClient)
 		pRaceInfo->m_RaceStartTick = m_pCharacter->m_StartTime;
 	}
 
-	if(m_pCharacter && m_pCharacter->IsPaused())
+	CPlayer* SnapPlayer = GameServer()->m_apPlayers[SnappingClient];
```
Comment (heinrich5991) [inline src/game/server/player.cpp:407]: `SnappingClient` can be -1 for demos. Naming convention, code style.

File: src/game/server/player.cpp [hunk 3]
```diff
@@ -404,7 +404,12 @@ void CPlayer::Snap(int SnappingClient)
...
+
+	if(m_pCharacter && m_pCharacter->IsPaused()
+		&& (GameServer()->GetDDRaceTeam(id) == GameServer()->GetDDRaceTeam(SnappingClient) 
+			|| m_ShowOthers 
+			|| (SnapPlayer->GetTeam() == TEAM_SPECTATORS || SnapPlayer->IsPaused())))
```
Comment (heinrich5991) [inline src/game/server/player.cpp:412]: Is there no function for this?

This looks wrong, it shows a spec tee to everyone if the tee has `m_ShowOthers` on?

Comment (edg-l) [reply src/game/server/player.cpp:412]: isn't that what showothers does?

File: src/game/server/player.cpp (hunk 2 above)
Comment (edg-l) [reply src/game/server/player.cpp:407]: funny tho, SnapPlayer name is used in character.cpp

Comment (edg-l) [reply src/game/server/player.cpp:407]: Some lines up you can clearly see `GameServer()->m_apPlayers[SnappingClient]->m_TimerType == TIMERTYPE_SIXUP`, and it doesn't look like it handles -1 either, I'm  confused

File: src/game/server/player.cpp (hunk 3 above)
Comment (edg-l) [reply src/game/server/player.cpp:412]: I didn't find any, 

ShowOthers description is: "Show players in other teams" so i think thats the intended way.

**Comment by edg-l** [comment]
Fixed and rebased

**Comment by heinrich5991** [comment]
Thanks!

bors r+

**Pull request merged**